Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -0,0 +1,107 @@
+/// Represents a MongoDB connection string.
+/// - SeeAlso: https://docs.mongodb.com/manual/reference/connection-string/
+public struct MongoConnectionString: Codable, LosslessStringConvertible {
```

Comment:
As we build up functionality for this type, we'll want to continually enable more of the connection string spec tests. For this PR in particular, we should be able to pass the subset of tests that only verify host and scheme information (I think this is only `valid-host_identifiers.json`). As part of that, we'll need to update the runner in `ConnectionStringTests.swift` to use `MongoConnectionString` instead of the old `ConnectionString` and comment out the assertions that we're not able to pass yet (e.g. anything checking the options or authentication information). We'll also need to update `shouldSkip` to skip anything we can't pass yet. As we implement more PRs, we can uncomment more assertions and unskip more tests.

Reply:
done